src/api/app/models/cloud/backend/upload_job.rb

```diff
@@ -32,8 +32,17 @@ def self.create(params)
         new(exception: exception.message)
       end
 
-      def self.all(user)
+      def self.find(job_id, user, options = {})
+        xml = ::Backend::Api::Cloud.upload_jobs([job_id])
+        return xml if options[:format] == :xml
+        OpenStruct.new(Xmlhash.parse(xml)['clouduploadjob'])
```

Comment:
You should return an ``UploadJob`` object and not an ``OpenStruct``

``
new(xml_object: OpenStruct.new(xml_hash))
``

See ``#all`` method!